modules/lang-groovy/src/main/java/org/elasticsearch/script/groovy/GroovyScriptEngineService.java

```diff
@@ -91,6 +92,11 @@
      */
     private final ClassLoader loader;
 
+    /**
+     * Ensures that the deprecation log entry for Groovy is only written on the first Groovy script compiled.
+     */
+    private AtomicBoolean isDeprecationLogged = new AtomicBoolean(false);
```

Comment:
I don't know that we want to limit it to only the first compilation. We may also want to warn on each use as well (search and execution so that users know where they're using them if it's hidden away in some existing code somewhere).

Specifically, if they are using file-based scripts, then we lose the other half of deprecation logging, which is the `Warning` header that gets sent back to HTTP clients starting in 5.0. I'd want to complain often and loudly. Since it's just like any other logger, they can just disable `deprecation.script.groovy` deprecation logging if they choose to do it.

----

If we do keep it, then this should be `final`.